lib/SIL/IR/SILValue.cpp

```diff
@@ -163,38 +136,9 @@ SILLocation SILValue::getLoc() const {
   if (auto *arg = dyn_cast<SILArgument>(*this)) {
     if (arg->getDecl())
       return RegularLocation(const_cast<ValueDecl *>(arg->getDecl()));
-
-    // Take one of the incoming operand locations if we have any.
-    SmallPtrSet<SILArgument*, 4> visitedSet;
-    auto inst = tryToGetArgumentInst(arg, visitedSet);
-    if (inst) return inst->getLoc();
-  }
-
-  // If this is something that lives in a function, fall back to the function
-  // loc.  Otherwise, e.g. a SILUndef, return a null location.
-  auto fn = Value->getFunction();
-  return fn ? fn->getLocation() : SILLocation((Expr*)nullptr);
-}
-
-/// Get a debug location for this value.
-SILDebugLocation SILValue::getDebugLocation() const {
```

Comment:
Just for reference: it might be nice to upstream these `tensorflow` branch changes. They were added as part of graph program extraction to improve SIL debug locations.